include/clover_field.h

```diff
@@ -198,6 +239,20 @@ namespace quda {
        @return Absolute minimum value
      */
     double abs_min(bool inverse = false) const;
+
+    virtual int full_dim(int d) const { return x[d]; }
+
+    /**
+      @brief Copy all contents of the field to a host buffer.
+      @param[in] the host buffer to copy to.
+    */
+    virtual void copy_to_buffer(void *buffer) const = 0;
```

Comment:
Is there any reason these can't become abstract members of the `LatticeField` instead of `CloverField`, `ColorSpinorField`, etc?

Reply:
Thanks for catching that.